scripts/generate_dump

```diff
@@ -1582,6 +1619,25 @@ collect_broadcom() {
 
     copy_from_masic_docker "syncd" "/var/log/diagrun.log" "/var/log/diagrun.log"
     copy_from_masic_docker "syncd" "/var/log/bcm_diag_post" "/var/log/bcm_diag_post"
+
+    # run 'hw-management-generate-dump.sh' script and save the result file
+    HW_DUMP_FILE=/usr/bin/hw-management-generate-dump.sh
```

Comment:
This PR should also cleanup the duplicate code in vendor specific functions, otherwise it will cause duplicate execution and increase run time.

Reply:
ignore, looks like being done only for broadcom. 
Should we move it to common code?
The code gets executed only when vendor provide the executable.

Reply:
Melanox and Cisco have implemented the function call hw-management-generate-dump.sh under ASIC specified (collect_mellanox() and collect_cisco_8000.  For Broadcom ASIC, we just follow the existing implemenation for avoid the duplicated function call.
```
    if [[ "$asic" = "mellanox" ]]; then
        collect_mellanox
    fi

    if [ "$asic" = "broadcom" ]; then
        collect_broadcom
    fi

    if [ "$asic" = "cisco-8000" ]; then
        collect_cisco_8000
    fi
```